flink-table/flink-table-runtime/src/main/java/org/apache/flink/table/runtime/operators/join/temporal/TemporalRowTimeJoinOperator.java

```diff
@@ -202,7 +202,7 @@ public void close() throws Exception {
      *     have been processed.
      */
     private long emitResultAndCleanUpState(long currentWatermark) throws Exception {
-        TemporalValue<Iterable<RowData>> leftRow;
+        TemporalValue<? extends Iterable<RowData>> leftRow;
```

Comment:
It would be great if we could get the types so this was `TemporalValue<Iterable<RowData>>` and remove the existential

Reply:
That's probably only possible by rewrapping the TemporalValue, right?
Not sure how to generalise that without creating too much boilerplate code
:(

Reply:
Thinking about it a bit more, maybe this can be done by parameterizing the
return value of deserializeValue which I added last minute

Reply:
I tried this and I believe, this is only possible if we remove `InternalKvState#getNavigator` and make this an internal implementation detail of `RocksDBListState` and `RocksDBValueState` by adding the `getAtOrBefore` and `getAtOrAfter` methods from their user-facing counterparts to these classes instead (or at least the navigator logic)

Reply:
nevermind - I provided a solution based on a wrapper class for `TemporalValue<Iterable<T>>`